### DISCUSSION TITLE
How to get nodes which represent `autoload` statement？

### DISCUSSION THREAD

**Discussion by Super-Xray** [ruby/prism#2696] [Q&A]
I'm working on using prism to derive statements like `autoload :Bar, "bar"` but I couldn't find which kind of `Node` contains this kind of statements. I had tried to search nodes definition which contains `autoload`  in the document and source codes and didn't find valid result.

**Comment by kddnewton** [comment]
Hi @Super-Xray — those are all `CallNode` nodes, because `autoload` isn't a keyword. To find them all, a visitor would be a good idea, as in:

```ruby
class AutoloadVisitor < Prism::Visitor
  attr_reader :autoloads

  def initialize(autoloads)
    @autoloads = autoloads
  end

  def visit_call_node(node)
    autoloads << node if node.name == :autoload
    super
  end
end

source = <<~RUBY
module Foo
  autoload :Bar1, "bar.rb",
  autoload :Bar2, "bar.rb"
  autoload :Bar3, "bar.rb"
end
RUBY

autoloads = []
Prism.parse(source).value.accept(AutoloadVisitor.new(autoloads))

p autoloads # [#<Prism::CallNode>, #<Prism::CallNode>, #<Prism::CallNode>]
```

you may want to additionally restrict it by checking if `node.receiver.nil?` and anything else.

**Reply by Super-Xray** [reply]
thank you very much
